kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/util.kt

```diff
@@ -99,6 +101,10 @@ internal fun KtSymbolWithMembers.declarations(): Sequence<KSDeclaration> {
 internal fun KtSymbolWithMembers.getAllProperties(): Sequence<KSPropertyDeclaration> {
     return analyze {
         this@getAllProperties.getMemberScope().getCallableSymbols()
+            .filter {
+                it.isVisibleInClass(this@getAllProperties as KtClassOrObjectSymbol) ||
```

Comment:
This one is `KtOverrideInfoProvider#isVisible`, which is checking if the given (possibly inherited) member symbol is visible in the given (possibly subtype) class. They're originally used by FIR checker, `FirNotImplementedOverrideChecker`, and even didn't care containing class membership.

I think your intention is rather `FirVisibilityChecker#isVisible` (note that that util has an option to `skipCheckForContainingClassVisibility` for simplicity!), which is wrapped as `KtVisibilityChecker#isVisible`.